Documentation: Kedro-Viz in VSCode extension (#93)
kedro-org/vscode-kedro: README.md

│           └── second_pipeline

```

> 

Reviewer: ```suggestion
To visualize your Kedro project using Kedro-Viz in Visual Studio Code, follow these steps:

**Open the Command Palette**: 
Press Cmd + Shift + P (on macOS) or Ctrl + Shift + P (on Windows/Linux).

**Run Kedro-Viz**:
Type `kedro: Run Kedro Viz` and select the command.

This will launch Kedro-Viz and display your pipeline visually within the extension.
```

## Visualisation with Kedro-Viz

> To visualized Kedro project within Kedro extension with the help of inbuilt pipeline visualisation tool Kedro-Viz.

Reviewer 2: ```suggestion
To visualize Kedro project within Kedro extension with the help of inbuilt pipeline visualisation tool Kedro-Viz.
```

> ## Visualisation with Kedro-Viz
> To visualize Kedro project within Kedro extension with the help of inbuilt pipeline visualisation tool Kedro-Viz.
> Use Command (`Cmd` + `Shift` + `P`) and choose `kedro: Run Kedro Viz`
> ![start kedro viz](assets/viz-vsc-start.gif)
> 
> 

Reviewer: ```suggestion
## Visualisation with Kedro-Viz
To visualize your Kedro project using Kedro-Viz in Visual Studio Code, follow these steps:
**Open the Command Palette**: 
Press Cmd + Shift + P (on macOS) or Ctrl + Shift + P (on Windows/Linux).
**Run Kedro-Viz**:
Type `kedro: Run Kedro Viz` and select the command.
This will launch Kedro-Viz and display your pipeline visually within the extension.

```

## Hover
Just hover your mouse over any `params:`, datasets or hit the command `Show or Focus Hover`
![hover](assets/lsp-hover.gif)

## Flowchart to Editor navigation

Clicking on a node in flowchart navigate to the corresponding node function in the code.

> ![schema validation](assets/viz-vsc-nav-function-node.gif)

Reviewer 2: ```suggestion
![schema validation](assets/viz-vsc-nav-function-node.gif)
```

I suppose `schema validation` is copied from somewhere

Clicking on a data node in flowchart navigate to the corresponding dataset yaml file.

> ![navigation to dataset](assets/viz-vsc-nav-data-node.gif)

Reviewer: To navigate between the flowchart and the code editor in Kedro-Viz:

1. Navigate to Node Functions:

Click on a node in the Kedro-Viz flowchart, and it will automatically navigate to the corresponding node function in your code.

2. Navigate to DataCatalog:

Clicking on a data node in the flowchart will open the corresponding dataset in the Data Catalog.

## Hover
Just hover your mouse over any `params:`, datasets or hit the command `Show or Focus Hover`
![hover](assets/lsp-hover.gif)

## To navigate between the flowchart and the code editor in Kedro-Viz:

Navigate to Node Functions:

> Click on a node in the Kedro-Viz flowchart, and it will automatically navigate to the corresponding node function in your code.

Reviewer 3: I'm pretty late to this, but I'm wondering: since the navigation functionality on the editor is `Cmd+Click` (see "Go to Definition from pipeline.py to configuration files" and "Go to Reference from configuration files to pipeline.py" above in this same README), and that clicking nodes has a different behavior on the Viz web application...

...should the navigation action for Viz on VS Code be `Cmd/Ctrl+Click` as well, for consistency?

cc @stephkaiser @iamelijahko

Reviewer 2: Not sure if I like the Cmd+click, unless hitting cmd show something different (like in editor it will turn text into underlined blue text). Either way it's inconsisent either with the IDE or vis web.

Reviewer 4: if I've understood correctly.. I think its okay to keep the 'clicking nodes' interaction on Viz because thats what users are familiar with and its consistent with how users currently interact with Viz. And I assume the Cmd+Click behaviour is also standard on VS code. I'd suggest to keep it like this, it makes the most sense to me for now.
